src/modules/usart.rs

```diff
@@ -0,0 +1,258 @@
+                    Loc3 => {
+                        self.init_gpio(Some((Port::E, 13)), Some((Port::E, 12)),
+                            None, None);
+                        usart::ROUTE_LOCATION_LOC3
+                    },
+                    Loc4 => {
+                        self.init_gpio(Some((Port::B, 7)), Some((Port::B, 8)),
+                            Some((Port::B, 13)), Some((Port::B, 14)));
+                        usart::ROUTE_LOCATION_LOC4
+                    },
+                    Loc5 => {
+                        self.init_gpio(Some((Port::C, 0)), Some((Port::C, 1)),
+                            Some((Port::B, 13)), Some((Port::B, 14)));
+                        usart::ROUTE_LOCATION_LOC5
+                    },
+                }
+            },
+            Config::Usart1(loc) => {
+                match loc {
+                    Loc0 => {
+                        self.init_gpio(Some((Port::C, 0)), Some((Port::C, 1)),
+                            Some((Port::B, 7)), Some((Port::B, 8)));
+                        usart::ROUTE_LOCATION_LOC0
+                    },
+                    Loc1 => {
+                        self.init_gpio(Some((Port::D, 0)), Some((Port::D, 1)),
+                            Some((Port::D, 2)), Some((Port::D, 3)));
+                        usart::ROUTE_LOCATION_LOC1
+                    },
+                    Loc2 => {
+                        self.init_gpio(Some((Port::D, 7)), Some((Port::D, 6)),
+                            Some((Port::F, 0)), Some((Port::F, 1)));
+                        usart::ROUTE_LOCATION_LOC2
+                    },
+                    _ => panic!("Invalid location for Usart1: {:?}", loc)
+                }
+            },
+            Config::Usart2(loc) => {
+                match loc {
+                    Loc0 => {
+                        self.init_gpio(Some((Port::C, 2)), Some((Port::C, 3)),
+                            Some((Port::C, 4)), Some((Port::C, 5)));
+                        usart::ROUTE_LOCATION_LOC0
+                    },
+                    Loc1 => {
+                        self.init_gpio(Some((Port::B, 3)), Some((Port::B, 4)),
+                            Some((Port::B, 5)), Some((Port::B, 6)));
+                        usart::ROUTE_LOCATION_LOC1
+                    },
+                    _ => panic!("Invalid location for Usart2: {:?}", loc)
+                }
+            },
+        };
+        self.usart.ROUTE = (self.usart.ROUTE & !usart::ROUTE_LOCATION_MASK) | location;
+    }
+
+    fn init_gpio(&mut self, rx: PortPin, tx: PortPin, clk: PortPin, cs: PortPin) {
+        // Enable clock for GPIO module (required for pin configuration)
+        cmu::clock_enable(cmu::Clock::GPIO, true);
+
+        // Configure GPIO pins for TX, RX, CLK and CS
+        match rx {
+            Some((port, pin)) => {
+                gpio::pin_mode_set(port, pin, gpio::Mode::PushPull, 1);
+                self.usart.ROUTE |= usart::ROUTE_TXPEN;
```

Comment:
could you reference self.usart.ROUTE once. I.e. build up a flag and set it after all matches.